cpp/benchmarks/hausdorff_benchmark.cpp

```diff
@@ -19,22 +19,20 @@
 #include <benchmarks/fixture/benchmark_fixture.hpp>
 #include <benchmarks/synchronization/synchronization.hpp>
 
+#include <cudf/detail/iterator.cuh>
 #include <cudf_test/column_wrapper.hpp>
 
-#include <thrust/iterator/constant_iterator.h>
-
 static void BM_hausdorff(benchmark::State& state)
 {
   int32_t num_points           = state.range(1) - 1;
   int32_t num_spaces_asked     = state.range(0) - 1;
   int32_t num_spaces           = std::min(num_points, num_spaces_asked);
   int32_t num_points_per_space = num_points / num_spaces;
 
-  auto counting_iter = thrust::counting_iterator<int32_t>();
-  auto zero_iter     = thrust::make_transform_iterator(counting_iter, [](auto idx) { return 0; });
+  auto zero_iter = cudf::detail::make_counting_transform_iterator(0, [](auto idx) { return 0; });
```

Comment:
This iterator always returns zero, so why not use `thrust::make_constant_iterator(0)`?

Reply:
Yes that's much better.